test/CMakeLists.txt

```diff
@@ -26,130 +13,153 @@ if (BUILD_TESTS)
     if (USE_Z3)
         target_link_libraries(testrunner ${Z3_LIBRARIES})
     endif()
+    if (WIN32 AND NOT BORLAND)
+        target_link_libraries(testrunner Shlwapi.lib)
+    endif()
 
     if (CMAKE_VERSION VERSION_EQUAL "3.16" OR CMAKE_VERSION VERSION_GREATER "3.16")
         target_precompile_headers(testrunner PRIVATE precompiled.h)
     endif()
 
     add_custom_target(copy_cfg ALL
+        ${CMAKE_COMMAND} -E copy_directory "${PROJECT_SOURCE_DIR}/cfg"
+            "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/cfg"
         COMMENT "Copying cfg files")
-    add_custom_command(
-        TARGET copy_cfg
-        COMMAND ${CMAKE_COMMAND} -E copy_directory "${PROJECT_SOURCE_DIR}/cfg"
-            "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/cfg")
-
-    if (WIN32 AND NOT BORLAND)
-        target_link_libraries(testrunner Shlwapi.lib)
-    endif()
-
     add_dependencies(testrunner copy_cfg)
-    add_dependencies(check testrunner cppcheck)
 
-    set(SKIP_TESTS "" CACHE STRING "A list of tests to skip")
+    if (LIBXML2_XMLLINT_EXECUTABLE)
+        # TODO: get rid of the copy
+        add_custom_target(checkcfg ${CMAKE_COMMAND} -E copy $<TARGET_FILE:cppcheck> ${CMAKE_SOURCE_DIR}
```

Comment:
We shouldn't copy files into the source directory. First, there is no clean up for this, and secondly, it can cause problems if its ran across two different builds in parallel.

Reply:
Unfortunately the `runtests.sh` awaits the binary there as that's where the Makefile build puts the file.